src/inject/dynamic-theme/adopted-style-manger.ts

```diff
@@ -0,0 +1,93 @@
+import {Theme} from '../../definitions';
+import {createStyleSheetModifier} from './stylesheet-modifier';
+import {forEach} from '../../utils/array';
+
+let adoptedSheetOverride = new WeakMap<CSSStyleSheet, CSSStyleSheet>();
+let nodes = [] as Array<ShadowRoot | Document>;
+let overrides = new WeakSet<CSSStyleSheet>();
+
+export function removeAdoptedStyleSheets() {
+    forEach(nodes, (node) => {
+        forEach(node.adoptedStyleSheets, (adoptedStyleSheet) => {
+            if (overrides.has(adoptedStyleSheet)) {
+                const newSheets = [...node.adoptedStyleSheets];
+                const existingIndex = newSheets.indexOf(adoptedSheetOverride.get(adoptedStyleSheet));
+                if (existingIndex >= 0) {
+                    newSheets.splice(existingIndex, 1);
+                }
+                node.adoptedStyleSheets = newSheets;
+            }
+        });
+    });
+    cleanAdoptedStyleSheets();
+}
+function cleanAdoptedStyleSheets() {
+    adoptedSheetOverride = new WeakMap<CSSStyleSheet, CSSStyleSheet>();
```

Comment:
do the cleanup (like `adoptedSheetOverrides.delete(sheet)` etc) in `destroy()` method of each manager